File: 30 Days of Code/Day 13 Abstract Classes/Day 13 Abstract Classes.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
#include <string>
using namespace std;
class Book{
    protected:
        string title;
        string author;
    public:
        Book(string t,string a){
            title=t;
            author=a;
        }
        virtual void display()=0;

};

// Write your MyBook class here
class MyBook : Book {
    //   Class Constructor
    //   Parameters:
    //   title - The book's title.
    //   author - The book's author.
    //   price - The book's price.
    //
    // Write your constructor here
    protected:
            int price;
         public:
            MyBook(string t, string a, int p) : Book (t, a) {
                this->price = p;
            }
    //   Function Name: display
    //   Print the title, author, and price in the specified format.
    //
    // Write your method here
        void display() override {
            cout << "Title: " << this->title << endl;
            cout << "Author: " << this->author << endl;
            cout << "Price: " << this->price << endl;
        }
    
};
// End class

int main() {
    string title,author;
    int price;
    getline(cin,title);
    getline(cin,author);
    cin>>price;
    MyBook novel(title,author,price);
    novel.display();
    return 0;
}

File: 30 Days of Code/Day 13 Abstract Classes/Day 13 Abstract Classes.cs
using System;
using System.Collections.Generic;
using System.IO;
abstract class Book
{
    
    protected String title;
    protected  String author;
    
    public Book(String t,String a){
        title=t;
        author=a;
    }
    public abstract void display();


}

class MyBook: Book {
    double price;
    public MyBook(string t, string a, double p) : base(t, a)
    {
        price = p;
    }

    public override void display()
    {
        Console.WriteLine(string.Format("Title: {0}", title));
        Console.WriteLine(string.Format("Author: {0}", author));
        Console.WriteLine(string.Format("Price: {0}", price));
    }
}

class Solution {
    static void Main(String[] args) {
      String title=Console.ReadLine();
      String author=Console.ReadLine();
      int price=Int32.Parse(Console.ReadLine());
      Book new_novel=new MyBook(title,author,price);
      new_novel.display();
    }
}